chain-core/src/property.rs

```diff
@@ -257,6 +257,8 @@ pub trait Settings {
 
     /// apply the Update to the Settings
     fn apply(&mut self, update: Self::Update) -> Result<(), Self::Error>;
+
+    fn tip(&self) -> &<Self::Block as Block>::Id;
```

Comment:
No need to commit to keeping a reference to tip inside the type. Consider returning by value.